src/components/views/settings/devices/DeviceDetails.tsx

```diff
@@ -87,15 +88,23 @@ const DeviceDetails: React.FC<Props> = ({
             </table>,
             ) }
         </section>
-        { !!onSignOutDevice && <section className='mx_DeviceDetails_section'>
+        <section className='mx_DeviceDetails_section'>
             <AccessibleButton
                 onClick={onSignOutDevice}
                 kind='danger_inline'
+                disabled={isSigningOut}
                 data-testid='device-detail-sign-out-cta'
             >
-                { _t('Sign out of this session') }
+                <span className='mx_DeviceDetails_signOutButtonContent'>
+                    { _t('Sign out of this session') }
+                    { isSigningOut && <>
+                    &nbsp;
```

Comment:
We could avoid the fragment by replacing `&nbsp;` with `gap` in `.mx_DeviceDetails_signOutButtonContent`